src/main/scala/org/constellation/concurrency/cuckoo/MutableCuckooFilter.scala

```diff
@@ -0,0 +1,67 @@
+package org.constellation.concurrency.cuckoo
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+
+import cats.effect.Sync
+import net.cinnom.nanocuckoo.NanoCuckooFilter
+import org.constellation.schema.snapshot.FilterData
+import cats.syntax.all._
+
+class MutableCuckooFilter[F[_], T](filter: NanoCuckooFilter)(implicit sync: Sync[F]) {
+
+  def insert(item: T)(implicit convert: T => String): F[Boolean] = sync.delay {
+    val converted = convert(item)
+    if (filter.insert(converted))
+      true
+    else {
+      filter.expand()
+      filter.insert(converted)
+    }
+  }
+
+  def delete(item: T)(implicit convert: T => String): F[Boolean] = sync.delay {
+    convert.andThen(filter.delete)(item)
+  }
+
+  def contains(item: T)(implicit convert: T => String): F[Boolean] = sync.delay {
+    convert.andThen(filter.contains)(item)
+  }
+
+  def getFilterData: F[FilterData] =
+    for {
+      capacity <- getCapacity
+      contents <- getFilterContents
+    } yield FilterData(contents, capacity)
+
+  def clear: F[Unit] =
+    for {
+      capacity <- getCapacity
+      contents <- MutableCuckooFilter(capacity).getFilterContents
+      _ <- setFilterContents(contents)
+    } yield ()
+
+  private def getCapacity: F[Long] = sync.delay {
+    filter.getCapacity
```

Comment:
why it's delayed? does it make a side effect?

Reply:
This method does't make any side effects, but I want to make it referentially transparent, since it may yield different results for the same input. Would `Sync[F].pure` cover that?

Reply:
I would use `pure` then :+1:

Reply:
After all the discussions in the slack, we keep the `delay` here.